diff -pruN gmusicbrowser.orig/plugins/lyrics.pm gmusicbrowser/plugins/lyrics.pm
--- gmusicbrowser.orig/plugins/lyrics.pm	2017-03-11 07:42:22.454322440 +0200
+++ gmusicbrowser/plugins/lyrics.pm	2017-03-11 07:57:38.551538276 +0200
@@ -73,18 +73,27 @@ my %Sites=	# id => [name,url,?post?,func
 	lyricwiki =>	[lyricwiki => 'http://lyrics.wikia.com/%a:%t',undef,
 			 sub {	return 0,'http://lyrics.wikia.com/'.$1 if $_[0]=~m#<span class="redirectText"><a href="/([^"]+)"#;
 				$_[0]=~s!.*<div class='lyricbox'>.*?((?:&\#\d+;|<br ?/>|</?[bi]>){5,}).*!$1!s; #keep only the "lyric box"
+				return $notfound if $_[0]=~m!This is a list of songs not being assigned to an artist yet!;#fix help out page displaying
 				return 0 if $_[0]=~m!&#91;&#46;&#46;&#46;&#93;(?:<br ?/>)*<i>!; # truncated lyrics : "[...]" followed by italic explanation => not auto-saved
 				return !!$1;
 			}],	
-	musixmatch =>   [musixmatch => 'https://www.musixmatch.com/lyrics/%a/%t', undef,
-                         sub {
-                            if ($_[0] =~ /<span id="lyrics-html"/) {
-                                $_[0] =~ s/.*<span id="lyrics-html".+?>(.+?)<\/span>.*/$1/s;
-                                $_[0] =~ s/[\r\n]/<br>/g;
-                            } else {
-                                $_[0] = $notfound;
-                            }
-                        }],
+	musixmatch =>   [musixmatch =>
+			sub { ::ReplaceFields($_[0], 'https://www.musixmatch.com/lyrics/%a/%t', sub {
+				$_[0]  =~ s/ (&|\*|\-) / /gr
+				   =~ s/(\'|\,|\.) / /gr
+				   =~ s/\(.*?\)//gr # remove comments in parenthesis: a.e. (Extended Version), (Chillout Mix), etc
+				   =~ s/^\s+|\s+$//gr
+				});
+			},
+			undef,
+			sub {
+				my $text;
+				if($_[0] =~ /"body":"(.*?)","/){
+					$text = $1 =~ s|\\n|<br>|gr =~ s|\\(["'])|$1|gr;
+				}
+				$_[0] = $text || $notfound;
+				return !!$text;
+			}],
        #lyricwikiapi => [lyricwiki => 'http://lyricwiki.org/api.php?artist=%a&song=%t&fmt=html',undef,
 	#	sub { $_[0]!~m#<pre>\W*Not found\W*</pre>#s }],
 	#azlyrics => [ azlyrics => 'http://search.azlyrics.com/cgi-bin/azseek.cgi?q="%a"+"%t"'],
